### ISSUE TITLE
Detect with a linter if the argument is not an ARN in AWS::CodePipeline::Pipeline

### ISSUE THREAD

**Issue by mariano-calandra-xp** [aws-cloudformation/cfn-lint#1393]
*cfn-lint version: cfn-lint 0.28.2*

*Description of issue.*
I was having really hard time debugging an error with CloudFormation syntax: when the CloudFormation try to deploy the stack it gave me an error like: "user:xyz/CloudFormationRole is not authorized to iam:passRole on resource".

In the first time I was going crazy searching for the error in IAM role, but after a stroll I realized that the error was here `RoleArn: !Ref CloudFormationRole`. 

```yml
- Name: Deploy
            Actions:
              - Name: CreateChangeSet
                ActionTypeId:
                  Category: Deploy
                  Owner: AWS
                  Provider: CloudFormation
                  Version: '1'
                InputArtifacts:
                  - Name: BuildArtifact
                Configuration:
                  ActionMode: CHANGE_SET_REPLACE
                  Capabilities: CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND
                  RoleArn: !Ref CloudFormationRole # ERROR HERE (next line is correct)
                  # RoleArn: !GetAtt CloudFormationRole.Arn
                  StackName: lambda-deploy-stack
                  TemplatePath: "BuildArtifact::package.yaml"
                  ChangeSetName: lambda-deploy-changeset
                RunOrder: 1
```

Would be useful if the linter could validate this type of error upfront. 
Here is an expanded version of YML https://gist.github.com/mariano-calandra-xp/8cb92afdd379aeed09a5df1cd2582c65

**Comment by PatMyron** [comment]
Agreed this is frustrating and it should be possible for the Linter to catch eventually if `CloudFormationRole` was created in the same template

In the meantime, I've reached out to the maintainer of this [Ref and GetAtt cheatsheet](https://theburningmonk.com/cloudformation-ref-and-getatt-cheatsheet/) about using the [`us-east-1` CloudFormation Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) to automate maintenance of new return values

It does have return value information for many resource types like [`AWS::IAM::Role`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html) that might help:

```
Resource Type  | Ref  | Fn::GetAtt
AWS::IAM::Role | Name | Arn, RoleId
```

For new types not in the cheatsheet yet, [each individual resource type's reference documentation has a return value section as well](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#aws-resource-iam-role-return-values)

**Comment by kddejong** [comment]
We do have some spec patching for checking on this.  The only issue I really have is `Configuration` drops into JSON in the spec.  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions.html#cfn-codepipeline-pipeline-stages-actions-configuration.  Which is okay but we are going to have to do a little bit more to get this working correctly.

**Comment by kddejong** [comment]
Sorry, huge delay in this one but we have a very easy way to do this now.
